board.cpp

```diff
@@ -0,0 +1,164 @@
+        std::cout << "choose the width of the field (between 5 and 10)" << std::endl;
+        std::cin >> width;
+    }
+    while ((5 > height) || (height > 10)){
+        std::cout << "choose the height of the field (between 5 and 10)" << std::endl;
+        std::cin >> height;
+    }
+    while ((3 > mineCount) || (mineCount > 8)){
+        std::cout << "choose the number of mines on the field (between 3 and 8)" << std::endl;
+        std::cin >> mineCount;
+    }
+    board.width = width;
+    board.height = height;
+    board.playerCount = 2;
+    board.players = new Player[board.playerCount]; //dynamic array since we don't know the number of players beforehand
+    for(int p = 0; p < board.playerCount; p++){
+        Player player;
+        player.mineCount = mineCount;
+        player.id = p;
+        if(p==1 && pvx == "pve"){ //set player 2 to be controlled by AI, may need to be refactored when support for more than 2 players is implemented
+            player.isAI = true;
+        }
+        board.players[p] = player;
+        }
+    return board;
+}
+
+//print board to console
+//may be depreciated if an UI is implemented in the future
+void printField(Board board, int perspective){ //no se yo tampoco entiendo nada
+    bool isPositionEnabled;
+    for (int y = 0; y <= board.height; y++){ //para cada fila
+        if (y == 0){ //si es la primera fila, imprimimos solo espacio
+            std::cout << "   ";
+        }
+        else{ //caso contrario, imprimimos el numero de la fila
+            std::cout << y;
+            for(int space = 0; space < (3 - std::to_string(y).length()); space++){
+                std::cout << " ";
+            };
+        }
+        for (int x = 1; x <= board.width; x++){ // seguido del numero de fila, imprimimos todas las posiciones de la fila
+            if (y == 0){ //si es la primera fila, imprimimos los numeros de las columnas
+                for(int space = 0; space < (3 - std::to_string(x).length()); space++){
+                    std::cout << " ";
+                };
+                std::cout << std::to_string(x);
+            }
+            else{ //en caso contrario, imprimimos el estado de la posicion (" " = no existe, "O" = posicion en juego/estado desconocido, "M" = mina del jugador en la posicion)
+                isPositionEnabled = true;
+                for(std::vector<Position>::iterator it = board.disabledPositions.begin(); it != board.disabledPositions.end(); it++){
+                    if(it->xpos == x && it->ypos == y){
+                        isPositionEnabled = false;
+                    }
+                }
+                bool mineInPos=false;
+                if (perspective != -1){ //shows the mines belonging to player
+                    for(std::vector<Mine>::iterator it = board.placedMines.begin(); it != board.placedMines.end(); it++){
+                        if(it->xpos == x && it->ypos == y){
+                            if(it->owner == perspective){

```

Comment:
#should glue conditions: `if ((it->xpos == x) && (it->pos == y) && (it->owner == perspective))`